public/app/features/alerting/unified/components/rule-editor/util.ts

```diff
@@ -152,7 +152,7 @@ export function getThresholdsForQueries(queries: AlertQuery[]) {
       const threshold = condition.evaluator.params;
 
       // "classic_conditions" use `condition.query.params[]` and "threshold" uses `query.model.expression`
-      const refId = condition.query.params[0] ?? query.model.expression;
+      const refId = condition.query?.params[0] ?? query.model.expression;
```

Comment:
Could we add a simple test that covers the issue?

Reply:
sure! let's do it

Reply:
done! instead of adding a new test I removed the `query` field for the `thresholdExpression` in the test.

Reply:
Could we remove `operator` and `reducer` too? These are not necessary for threshold

Reply:
done!